src/core/mergefilters.c

```diff
@@ -263,28 +265,30 @@ static const VSFrameRef *VS_CC mergeGetFrame(int n, int activationReason, void *
                 const uint8_t *srcp2 = vsapi->getReadPtr(src2, plane);
                 uint8_t * VS_RESTRICT dstp = vsapi->getWritePtr(dst, plane);
 
-                void (*func)(const void *, const void *, void *, union vs_merge_weight, unsigned);
+                void (*func)(const void *, const void *, void *, union vs_merge_weight, unsigned) = 0;
                 union vs_merge_weight weight;
 
 #ifdef VS_TARGET_CPU_X86
-                if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
-                    func = vs_merge_byte_sse2;
-                else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
-                    func = vs_merge_word_sse2;
-                else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
-                    func = vs_merge_float_sse2;
-                else
-                    continue;
-#else
-                if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
-                    func = vs_merge_byte_c;
-                else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
-                    func = vs_merge_word_c;
-                else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
-                    func = vs_merge_float_c;
-                else
-                    continue;
+                if (d->cpulevel >= VS_CPU_LEVEL_SSE2) {
+                    if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
+                        func = vs_merge_byte_sse2;
+                    else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
+                        func = vs_merge_word_sse2;
+                    else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
+                        func = vs_merge_float_sse2;
+                }
 #endif
+                if (!func) {
+                    if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
+                        func = vs_merge_byte_c;
+                    else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
+                        func = vs_merge_word_c;
+                    else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
+                        func = vs_merge_float_c;
+                }
+
+                if (!func)
```

Comment:
This redundancy could be avoided by moving the `continue` as `else` condition into the block above.  
(happens again)